pkg/services/provisioning/orgs/orgs.go

```diff
@@ -0,0 +1,96 @@
+package orgs
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/grafana/grafana/pkg/bus"
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/models"
+)
+
+// Provision scans a directory for provisioning config files
+// and provisions the orgs in those files.
+func Provision(ctx context.Context, configDirectory string) error {
+	provisioner := newOrgProvisioner(log.New("provisioning.orgs"))
+	return provisioner.applyChanges(ctx, configDirectory)
+}
+
+// OrgProvisioner is responsible for provisioning orgs based on
+// configuration read by the `configReader`
+type OrgProvisioner struct {
+	log         log.Logger
+	cfgProvider *configReader
+}
+
+func newOrgProvisioner(log log.Logger) OrgProvisioner {
+	return OrgProvisioner{
+		log:         log,
+		cfgProvider: &configReader{log: log},
+	}
+}
+
+func (provisioner *OrgProvisioner) apply(ctx context.Context, cfg *configs) error {
+	if err := provisioner.deleteOrgs(ctx, cfg.DeleteOrgs); err != nil {
+		return err
+	}
+
+	for _, org := range cfg.Orgs {
+		cmd := &models.GetOrgByIdQuery{Id: org.Id}
+		err := bus.Dispatch(ctx, cmd)
+		if err != nil && !errors.Is(err, models.ErrOrgNotFound) {
+			return err
+		}
+
+		if errors.Is(err, models.ErrOrgNotFound) {
+			provisioner.log.Info("inserting org from configuration ", "id", org.Id, "name", org.Name)
+			insertCmd := createInsertCommand(org)
+			if err := bus.Dispatch(ctx, insertCmd); err != nil {
```

Comment:
where the bus removal, we would need to inject org/preferences service here and remove all the buses.